src/main/java/model/CarRandomNumberGenerator.java

```diff
@@ -0,0 +1,12 @@
+package model;
+
+import java.util.Random;
+
+public class CarRandomNumberGenerator implements RandomNumberGenerator {
+    Random random = new Random();

```

Comment:
테스트코드를 위해서 랜덤을 분리했으니 외부에서 주입하는 방식도 좋을 것 같아. 찾아보니 모킹 프레임워크(Mockito)도 있다고 하는데 이건 잘 모르겠네

```suggestion
      private final Random random;

    public CarRandomNumberGenerator(Random random) {
        this.random = random;
    }
```

Reply:
좋은 아이디어이지만, `CarRandomNumberGenerator`는 생성자를 닫아두어도 충분히 제 기능을 하는 것 같아서, 이번에는 그대로 둘 것 같아요. 그러나 요구사항들이 늘어난다면 제가 PR에도 이야기했던 다형성을 잘 살릴 수 있는 좋은 기회가 될 것 같으므로 그 때는 말씀하신 대로 생성자를 열어두거나, interface를 오버라이딩하는 다른 구현체를 만들어볼 것 같아요.

사실 "생성자를 여는 것" 과 같은 제안과 "인터페이스를 선언해 여러 구현체 클래스를 만드는 것" 은 둘 다 다형성을 위한 좋은 방법이지 않을까 해요. 저는 이미 후자를 사용해 이를 이루었으니 전자를 이번에는 당장 수용하지는 않으려고 하는 것 정도라 생각해요.

이 경우라면 생성자를 두지 않아도 요구사항을 고려했을 때 충분히 괜찮은 코드일 것이라 생각되시나요? 아니면 다른 제가 놓친 장점도 있을까요?

Reply:
당연히 좋은 코드라고 생각해요! 다만 그냥 하나의 방법을 제시했을 뿐..ㅎ